docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst

```diff
@@ -1,25 +1,430 @@
+
+After the update, it is recommended to check if any new warnings are raised when starting the daemon.
+
+To do so you can use the `-t` flag:
+
+.. code-block:: shell-session
+   :caption: Shell example command to check bareos configuration
+
+   root@host:~# bareos-fd -t -u root -g bareos
+   root@host:~# bareos-sd -t -u bareos -g bareos
+   root@host:~# bareos-dir -t -u bareos -g bareos
+
+
+Depending of the operating system and its configuration, you will have to restart the daemons.
+Use your operating system command to do so.
+
+
+.. code-block:: shell-session
+   :caption: Shell command to restart all bareos daemon with systemd on Linux
+
+   root@host:~# systemctl restart bareos-director bareos-storage bareos-filedaemon
+   root@host:~# systemctl status bareos-director bareos-storage bareos-filedaemon
+
+
+.. code-block:: shell-session
+   :caption: Shell command to restart all bareos daemon with service on FreeBSD
+
+   root@host:~# service bareos-dir restart
+   root@host:~# service bareos-fd restart
+   root@host:~# service bareos-sd restart
+
+
+
+.. _section-UpgradeMajor:
+
+Upgrading Bareos to a new major release
+---------------------------------------
+
+.. index::
+   single: Upgrade latest major version
+   pair: Bareos; Upgrade
+   pair: Upgrade; Major
+
+In most cases, a Bareos major upgrade can be achieve by:
+
+- Add new major repository (subscription only).
+- Package upgrade of the distribution.
+- Database schema upgrade with helper scripts (if schema was changed).
+- Configuration review to cleanup deprecated or removed parameters.
+- Review of home made scripts and manage their adaptation in case of changes.
+
+It is generally sufficient to upgrade directly to the latest release, without having to install any intermediate releases.
+However, it is required to read the release notes of all intermediate releases.
+
+One exception is when using a |mysql| Bareos catalog,
+which have been removed with Bareos :sinceVersion:`21.0.0: MySQL backend removed`.
+Therefore you first have to upgrade to Bareos 20 and migrate the |mysql| into a |postgresql| Bareos Catalog, see :ref:`section-MigrationMysqlToPostgresql`.
+
+
+Prepare the upgrade
+~~~~~~~~~~~~~~~~~~~
+
+If you not have already done those steps, please refer to instructions in :ref:`section-UpdateUpgradePreamble`.
+
+.. warning::
+
+   If you use any third party plugins, you should check and test their functionalities with the new major version beforehand.
+
+.. _sectionUgradeMajorRepository:
+
+Upgrade the Bareos download repositories
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This does only apply for subscription repositories h(ttps://download.bareos.com/bareos/release/).
+The community repository (https://download.bareos.org/current) will always contain the latest build of the most recent stable branch.
+
+- First remove the existing Bareos repository definitions, by either removing the definition file(s) or by using your package manager.
```

Comment:
Is this really required? The package manage should detect the latest version anyway and the `add_bareos_repository.sh` script overwrites the existing repository anyway (on all platforms?).

Reply:
I've been hit at least on one platform, I will try to reproduce it again.

Reply:
```
 bash add_bareos_repositories.sh
Adding repository 'bareos' .............................................................................................................................................................................[error]
Repository named 'bareos' already exists. Please use another alias.